services/src/main/java/org/fao/geonet/api/records/MetadataApi.java

```diff
@@ -524,40 +524,56 @@ public RelatedResponse getRelated(
         @ApiResponse(code = 403, message = ApiParams.API_RESPONSE_NOT_ALLOWED_CAN_VIEW)
     })
     @ResponseBody
-    public FeatureResponse getFeatureCatalog(
+    public FeatureResponse getFeatureCatalog (
         @ApiParam(
             value = API_PARAM_RECORD_UUID,
             required = true)
         @PathVariable
             String metadataUuid,
-        HttpServletRequest request) throws Exception {
+        HttpServletRequest request) throws ResourceNotFoundException {
 
         RelatedItemType[] type = {RelatedItemType.fcats};
 
         FeatureResponse response = new FeatureResponse();
 
         Map<String, String[]> decodeMap = new HashMap<>();
 
-        RelatedResponse related = getRelated(metadataUuid, type, 0, 100, request);
-
-        if (related.getFcats() != null) {
-            for (AttributeTable.Element element : related.getFcats().getItem().get(0).getFeatureType().getAttributeTable().getElement()) {
-                if (element.getCode() != null && !element.getCode().trim().equals("")) {
-                    if (!decodeMap.containsKey(element.getCode())) {
-                        String[] decodedValues = {element.getName(), element.getDefinition()};
-                        decodeMap.put(element.getCode(), decodedValues);
-                    }
-                } else {
-                    if (!decodeMap.containsKey(element.getName())) {
-                        String[] decodedValues = {element.getName(), element.getDefinition()};
-                        decodeMap.put(element.getName(), decodedValues);
+        try {
+            RelatedResponse related = getRelated(metadataUuid, type, 0, 100, request);
+
+            if (isIncludedAttributeTable(related.getFcats())) {
+                for (AttributeTable.Element element : related.getFcats().getItem().get(0).getFeatureType().getAttributeTable().getElement()) {
+                    if (element.getCode() != null && !element.getCode().trim().equals("")) {
```

Comment:
Can be improved with `StringUtils.isNotEmpty`

Reply:
`isNotBlank`